File: sim-be/src/main/java/com/sim/backend/service/ProductService.java
package com.sim.backend.service;

import java.util.List;

import com.sim.backend.dto.request.CreateProductRequest;
import com.sim.backend.model.Product;

public interface ProductService {
    List<Product> getAllProducts();

    Product getProductById(Long id);

    Product createProduct(CreateProductRequest request);

    Product updateProduct(Long id, CreateProductRequest request);

    Product deleteProduct(Long id);
}

File: sim-fe/src/services/productService.js
import api from "@/lib/axios";

export const productService = {
    getAllProducts: async () => {
        const res = await api.get("/products");
        return res.data;
    },
    getProductById: async (id) => {
        const res = await api.get(`/products/${id}`);
        return res.data;
    },
    createProduct: async (data) => {
        const res = await api.post("/products", data);
        return res.data;
    },
    updateProduct: async (id, data) => {
        const res = await api.put(`/products/${id}`, data);
        return res.data;
    },
    deleteProduct: async (id) => {
        const res = await api.delete(`/products/${id}`);
        return res.data;
    }
}